### ISSUE TITLE
XGLM Bug: break: module 'torch.utils' has no attribute 'checkpoint'

### ISSUE THREAD

**Issue by mrseeker** [huggingface/transformers#15422]
## Environment info

- `transformers` version: 4.17.0.dev0
- Platform: Linux-4.9.253-tegra-aarch64-with-glibc2.29
- Python version: 3.8.10
- PyTorch version (GPU?): 1.10.2 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help
@patil-suraj 

## Information

Model I am using (Bert, XLNet ...): XGLMForCausalLM

The problem arises when using:
* [X] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [X] my own task or dataset: (give details below)

## To reproduce

1. I am trying to start training on a dataset using a converted model. 
2. As soon as I start training using the script `run_clm.py` (found in the examples folder) I get the following error back:
```
break: module 'torch.utils' has no attribute 'checkpoint'
```
Cause might be related to this: https://discuss.pytorch.org/t/attributeerror-module-torch-utils-has-no-attribute-checkpoint/101543/6

When adding the following line in the modeling_xglm.py, the error does not appear and training starts:

```
from torch.utils.checkpoint import checkpoint
```

## Expected behavior

Training would start without any issues.

**Comment by patil-suraj** [comment]
Hi,
Could you please share a code snippet or colab so I could reproduce this issue? Thanks

**Comment by mrseeker** [comment]
I can't share my dataset, but I used this one as the basis for the trainer:

https://github.com/huggingface/transformers/blob/master/examples/pytorch/language-modeling/run_clm.py

**Comment by patil-suraj** [comment]
Thank you for reporting this. Fixed in #15427
